### PR TITLE
Add Optional HTTPS Support For S3 Website

### PR DESCRIPTION
**Pull request by DaveOsment**

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -171,6 +185,58 @@ resources:
...
+          !Ref S3BucketPublicBucket
+        PolicyDocument: 
+          Statement:
+            - 
+              Action: 
+                - "s3:GetObject"
+              Effect: "Allow"
+              Resource: "arn:aws:s3:::${self:custom.config.public_bucket_name}/*"
+              Principal:
+                AWS: !Join [" ", ["arn:aws:iam::cloudfront:user/CloudFront Origin Access Identity", !Ref EphemeraCloudfrontOriginAccessIdentity]]
```
Comment (Sam-Martin) [inline serverless.yml:204]: Will this work @DaveOsment ? Won't it just have two ARNs separated by a space? is "CloudFront Origin Access Identity" even valid? I haven't use this functionality before.

**Comment by Sam-Martin** [comment]
I'm not 100% sure I want to add cloudfront on to Ephemera by default yet. As you can access the s3 bucket via https already on its default URL. Let me have a think about it @DaveOsment

File: serverless.yml (hunk 1 above)
Comment (DaveOsment) [reply serverless.yml:204]: When AWS auto-generated this for me, I asked exactly the same question myself and did a double-take! 

Upon testing, it creates the following entry in the S3 Bucket Policy:

```"Principal": {
                "AWS": "arn:aws:iam::cloudfront:user/CloudFront Origin Access Identity <Redacted>"
            },
```

This is permitting me to access Ephemera via CloudFront under test.

**Comment by DaveOsment** [comment]
CloudFront is disabled by default (The CloudFormation condition takes care of that, and only creates the CloudFront distribution and the relevant OAI when HTTPS is enabled) but I appreciate the rationale for not adding complexity where not strictly required to a project! :)

**Pull request merged**